app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/HomeActivity.java

```diff
@@ -1,18 +1,77 @@
 package com.udacity.nanodegree.blooddonation.ui.home;
 
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
 import android.os.Bundle;
 import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
 
+import com.google.firebase.auth.FirebaseAuth;
 import com.udacity.nanodegree.blooddonation.R;
 import com.udacity.nanodegree.blooddonation.base.BaseActivity;
+import com.udacity.nanodegree.blooddonation.ui.login.view.UserLoginActivity;
 
 /**
  * Created by riteshksingh on Apr, 2018
  */
 public class HomeActivity extends BaseActivity {
+
+    int REQUEST_CODE_LOCATION = 1 ;
+    int REQUEST_CODE_SMS = 2;
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_home);
+
+        locationPermissionCheck();
+        requestSmsPermission();
+    }
+
+    //Function to check and request permission for location
+    public void locationPermissionCheck() {
```

Comment:
try to improve function names with a little more verbosity requestPermissionIfNotPresent()
can we get rid of empty else cases or do they have a use?